awx_collection/plugins/modules/tower_inventory_source.py

```diff
@@ -190,23 +197,34 @@ def main():
     name = module.params.get('name')
     new_name = module.params.get('new_name')
     inventory = module.params.get('inventory')
+    organization = module.params.get('organization')
     source_script = module.params.get('source_script')
     credential = module.params.get('credential')
     source_project = module.params.get('source_project')
     state = module.params.get('state')
 
     # Attempt to look up inventory source based on the provided name and inventory ID
-    inventory_id = module.resolve_name_to_id('inventories', inventory)
-    inventory_source = module.get_one('inventory_sources', **{
+    org_id = module.resolve_name_to_id('organizations', organization)
+    inventory_object = module.get_one('inventories', **{
+        'data': {
+            'name': inventory,
+            'organization': org_id,
```

Comment:
wouldn't this be a problem in the event that the user didn't provide `organization`? In that case you get `None` for this, but in the query params I think that would cause it to filter for inventories with a null organization, which is probably 0.